packages/fxa-auth-server/lib/routes/oauth/authorization.js

```diff
@@ -129,16 +129,26 @@ module.exports = ({ log, oauthDB, config }) => {
     // Clients must use a single specific redirect_uri,
     // but they're allowed to not provide one and have us fill it in automatically.
     payload.redirect_uri = payload.redirect_uri || client.redirectUri;
-    if (payload.redirect_uri !== client.redirectUri) {
-      log.debug('redirect.mismatch', {
-        param: payload.redirect_uri,
-        registered: client.redirectUri,
-      });
+
+    // Starting in train-248, FxA supports a comma separated list of redirect urls.
+    // However, the client redirectUri must be an exact match, pattern matching
+    // is not supported.
+    const redirectUris = client.redirectUri.split(',');
+
+    const validUri = redirectUris.some((uri) => {
+      if (uri === payload.redirect_uri) {
```

Comment:
If there is no `redirectUri` passed in as part of the request, this would be comparing a single URI to potentially a list of URIs and this function would throw an error.

Also what if it's neither passed in with the request nor in the DB (i.e. `null` in the DB)?

Reply:
Good catch, I'll add some checks around these cases.